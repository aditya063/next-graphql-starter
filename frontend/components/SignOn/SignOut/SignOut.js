import { Mutation } from 'react-apollo';
import * as query from './SignOut.query';
import * as Styled from './SignOut.style';

const SignOut = React.memo(() => {
  const handleClickBtn = async (signOut, client) => {
    await signOut();
    await client.resetStore();
  };

  const handleError = error => error;

  return (
    <Mutation mutation={query.SIGN_OUT_MUTATION} onError={handleError}>
      {(signOut, { loading, client }) => (
        <Styled.redBtn
          disabled={loading}
          aria-busy={loading}
          onClick={() => handleClickBtn(signOut, client)}
        >
          Sign Out
        </Styled.redBtn>
      )}
    </Mutation>
  );
});

export default SignOut;
