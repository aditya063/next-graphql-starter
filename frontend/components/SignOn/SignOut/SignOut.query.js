import gql from 'graphql-tag';

export const SIGN_OUT_MUTATION = gql`
  mutation SIGN_OUT_MUTATION {
    signOut
  }
`;

export const MOCK_SIGN_OUT_MUTATION = {
  request: {
    query: SIGN_OUT_MUTATION
  },
  result: {
    data: {
      signOut: true
    }
  }
};
