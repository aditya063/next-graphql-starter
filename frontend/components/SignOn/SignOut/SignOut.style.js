import { redBtn } from '../../../styles/button.style';

export { redBtn };
