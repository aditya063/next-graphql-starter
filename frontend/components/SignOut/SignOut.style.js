export { purpleLinkBtn as signOutBtn } from '../Styled/button.style';
