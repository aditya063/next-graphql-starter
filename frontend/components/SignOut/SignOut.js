import { ApolloConsumer, Mutation } from 'react-apollo';

import { SIGN_OUT_MUTATION } from '../../graphql/queries';
import * as sc from './SignOut.style';

const SignOut = React.memo(() => {
  const handleClickBtn = async (signOut, client) => {
    await signOut();
    await client.resetStore();
  };

  const handleError = error => error;

  const handleCompleted = apolloClient =>
    apolloClient.writeData({ data: { isLoggedIn: false } });

  return (
    <ApolloConsumer>
      {apolloClient => (
        <Mutation
          mutation={SIGN_OUT_MUTATION}
          onError={handleError}
          onCompleted={() => handleCompleted(apolloClient)}
        >
          {(signOut, { loading, client }) => (
            <sc.signOutBtn
              disabled={loading}
              aria-busy={loading}
              onClick={() => handleClickBtn(signOut, client)}
            >
              Sign Out
            </sc.signOutBtn>
          )}
        </Mutation>
      )}
    </ApolloConsumer>
  );
});

export default SignOut;
