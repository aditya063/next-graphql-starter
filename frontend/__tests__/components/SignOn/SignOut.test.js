import React from 'react';
import { shallow } from 'enzyme';
import SignOut from '../../../components/SignOn/SignOut/SignOut';

describe('SignOut', () => {
  let mockProps;
  let wrapper;

  beforeEach(() => {
    jest.resetAllMocks();
    mockProps = {};
    wrapper = shallow(<SignOut {...mockProps} />, {
      disableLifecycleMethods: true
    });
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
