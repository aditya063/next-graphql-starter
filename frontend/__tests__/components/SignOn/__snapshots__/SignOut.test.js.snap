// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignOut matches snapshot 1`] = `
<Mutation
  errorPolicy="all"
  mutation={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "SIGN_OUT_MUTATION",
          },
          "operation": "mutation",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "signOut",
                },
                "selectionSet": undefined,
              },
            ],
          },
          "variableDefinitions": Array [],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 48,
        "start": 0,
      },
    }
  }
  onError={[Function]}
>
  <Component />
</Mutation>
`;
